src/server/scripts/Outland/zone_blades_edge_mountains.cpp

```diff
@@ -33,75 +33,164 @@ EndContentData */
+        void SpellHit(Unit* caster, SpellInfo const* spell) override
         {
-            if (spellInfo->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
+            if (spell->Id == SPELL_ARTILLERY_ON_THE_WRAP_GATE)
             {
-                Player* plr = caster->GetCharmerOrOwnerPlayerOrPlayerItself();
-                if (!plr)
-                    return;
-
-                timer = 1;
                 count++;
 
+                if (count >= 1)
+                {
+                    if (Player* player = caster->GetCharmerOrOwnerPlayerOrPlayerItself())
+                        PlayerGUID = player->GetGUID();
+
+                    CannonGUID = caster->GetGUID();
+                    PartyTime = true;
+                    PartyTimer = 3000;
+                }
+
+                if (count >= 3)
+                    me->SummonGameObject(GO_FIRE, me->GetPositionX(), me->GetPositionY(), me->GetPositionZ(), 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 130);
+
                 if (count >= 6)
                 {
-                    if (me->GetEntry() == NPC_SOUTH_GATE)
-                        plr->KilledMonsterCredit(NPC_SOUTH_GATE_CREDIT, 0);
-                    else
-                        plr->KilledMonsterCredit(NPC_NORTH_GATE_CREDIT, 0);
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, PlayerGUID))
+                    {
+                        if (GetClosestCreatureWithEntry(me, NPC_SOUTH_GATE, 200.0f))
+                            player->KilledMonsterCredit(NPC_SOUTH_GATE_CREDIT, TRIGGERED_NONE);
+                        
+                        else if (GetClosestCreatureWithEntry(me, NPC_NORTH_GATE, 200.0f))
+                            player->KilledMonsterCredit(NPC_NORTH_GATE_CREDIT, TRIGGERED_NONE);
+                        // complete quest part
+                        if (Creature* bunny = GetClosestCreatureWithEntry(me, NPC_EXPLOSION_BUNNY, 200.0f))
+                            bunny->CastSpell(nullptr, SPELL_EXPLOSION, TRIGGERED_NONE);
+                    }
 
+                    me->SummonGameObject(GO_BIG_FIRE, me->GetPositionX(), me->GetPositionY(), me->GetPositionZ(), 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 60);
                     Reset();
                 }
             }
+
+            return;
         }
 
-        void UpdateAI(uint32 diff)
+        void JustSummoned(Creature* summoned) override
+        {
+            if (summoned->GetEntry() == NPC_FEL_IMP)
+                summoned->CastSpell(summoned, SPELL_IMP_AURA, true);
+            
+            else if (summoned->GetEntry() == NPC_HOUND)
+                summoned->CastSpell(summoned, SPELL_HOUND_AURA, true);
+            
+            summoned->UpdateGroundPositionZ(x, y, z);
+
+            if (Creature* cannon = ObjectAccessor::GetCreature(*me, CannonGUID))
+                summoned->AI()->AttackStart(cannon);
+        }
+
+        void UpdateAI(uint32 diff) override
         {
-            if (timer)
+            if (PartyTime)
```

Comment:
Please, instead of timers use EventMap. Timers are obsolete. Better script with events instead ;)

Reply:
You still haven't changed this to eventmap